addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/discovery/ZoneMinderServerDiscoveryService.java

```diff
@@ -0,0 +1,278 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.zoneminder.discovery;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.Socket;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.net.util.SubnetUtils;
+import org.apache.commons.net.util.SubnetUtils.SubnetInfo;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.zoneminder.ZoneMinderConstants;
+import org.openhab.binding.zoneminder.handler.ZoneMinderServerBridgeHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Class ZoneMinderServerDiscoveryService.
+ *
+ * @author Martin S. Eskildsen
+ */
+public class ZoneMinderServerDiscoveryService extends AbstractDiscoveryService {
+
+    /** The logger. */
+    private final Logger logger = LoggerFactory.getLogger(ZoneMinderServerDiscoveryService.class);
+
+    /**
+     * Instantiates a new zone minder server discovery service.
+     *
+     * @throws IllegalArgumentException the illegal argument exception
+     */
+    public ZoneMinderServerDiscoveryService() throws IllegalArgumentException {
+        super(ZoneMinderServerBridgeHandler.SUPPORTED_THING_TYPES, 10);
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.eclipse.smarthome.config.discovery.AbstractDiscoveryService#getSupportedThingTypes()
+     */
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return ZoneMinderServerBridgeHandler.SUPPORTED_THING_TYPES;
+    }
+
+    // Background discovery disabled for now!
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        /*
+         * scheduler.schedule(new Runnable() {
+         *
+         * @Override
+         * public void run() {
+         * discoverZoneMinderServer();
+         * }
+         * }, 0, TimeUnit.MILLISECONDS);
+         */ }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.eclipse.smarthome.config.discovery.AbstractDiscoveryService#startScan()
+     */
+    @Override
+    protected void startScan() {
+        // discoverZoneMinderServer();
```

Comment:
hm, startScan should do something, otherwise the whole discovery service is useless?